storage/innobase/btr/btr0cur.cc

```diff
@@ -3697,8 +3697,11 @@ btr_cur_optimistic_update(
 	*offsets = rec_get_offsets(rec, index, *offsets, index->n_core_fields,
 				   ULINT_UNDEFINED, heap);
 #if defined UNIV_DEBUG || defined UNIV_BLOB_LIGHT_DEBUG
+	/* Blob pointer can be null if we ran out of space while
+	allocating the extern page. Relax the assertion during
+	rollback as well */
 	ut_a(!rec_offs_any_null_extern(rec, *offsets)
-	     || thr_get_trx(thr) == trx_roll_crash_recv_trx);
+	     || thr_get_trx(thr)->in_rollback);
```

Comment:
A better comment would be as follows:
```
	/* Blob pointer can be null if InnoDB was killed or
	ran out space while allocating a page. */
```
If you want to mention relaxing an assertion, you can do it in the commit message.